backtrace-library/src/main/java/backtraceio/library/services/BacktraceNativeCrashResponseListener.java

```diff
@@ -0,0 +1,127 @@
+        if (backtraceResult.status == BacktraceResultStatus.Ok) {
+            // Get the minidump out
+            BacktraceReport report = backtraceResult.getBacktraceReport();
+            if (report != null) {
+                List<String> attachments = report.attachmentPaths;
+                for (String attachment : attachments) {
+                    if (attachment.endsWith(".dmp")) {
+                        moveToCompleted(attachment);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    public static void UploadMinidumps(BacktraceBase client, String databasePath) {
+        if (client == null) {
+            BacktraceLogger.e(LOG_TAG, "Backtrace client can't be null");
+            return;
+        }
+        if (databasePath == "") {
+            BacktraceLogger.e(LOG_TAG, "Database path cannot be null");
+        }
+        if (instance == null) {
+            instance = new BacktraceNativeCrashResponseListener();
+        }
+        Thread sender = new Thread(() -> {
+            List<File> minidumps = scanForNewMinidumps(databasePath);
+            if (minidumps.size() > 0) {
+                for (File minidump: minidumps) {
+                    BacktraceReport report = createBacktraceReport(minidump, databasePath);
+                    client.send(report, instance);
+                }
+            }
+        });
+        sender.run();
+    }
+
+    private static List<File> scanForNewMinidumps(String databasePath) {
+        List<File> minidumps = new ArrayList<>();
+        FilenameFilter minidumpFilter = (dir, name) -> name.endsWith(".dmp");
+        String[] newFiles = new File(databasePath + "/new").list(minidumpFilter);
+        String[] pendFiles = new File(databasePath + "/pending").list(minidumpFilter);
+        if (newFiles != null) {
+            for (String file : newFiles) {
+                minidumps.add(new File(databasePath + "/new/" + file));
+            }
+        }
+        if (pendFiles != null) {
+            for (String file : pendFiles) {
+                minidumps.add(new File(databasePath + "/pending/" + file));
+            }
+        }
+        return minidumps;
+    }
+
+    private static BacktraceReport createBacktraceReport(File minidump, String databasePath) {
+        BacktraceReport report = new BacktraceReport("");
+        report.attachmentPaths.add(minidump.getAbsolutePath());
+        String attachmentPath = databasePath + "/attachments/"
+                + minidump.getName().replace(".dmp", "");
+        File attachmentsDir = new File(attachmentPath);
+        if (attachmentsDir.isDirectory()) {
+            for (String file: attachmentsDir.list()) {
+                report.attachmentPaths.add(attachmentsDir.getAbsolutePath() + "/" + file);
+            }
+        }
+        return report;
+    }
+
+    private void moveToCompleted(String minidumpPath) {
+        String outputPath = minidumpPath.replace("/new/", "/completed/")
```

Comment:
extract all of these `new` `completed` `pending` to consts

Reply:
Done.